src/redux/city/city.js

```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+import getAllCities from './city-helper';
+
+const initialState = {
+  all: [],
+};
+
+const citySlice = createSlice({
+  name: 'city',
+  initialState,
+  reducers: {
+    getAll: (state, action) => {
+      state.all = action.payload;
+    },
+  },
+});
+
+export const { getAll } = citySlice.actions;
+
+export const getAllCitiesAsync = () => async (dispatch) => {
+  const data = await getAllCities();
+  dispatch(getAll(data));
+};
```

Comment:
- [ ] To prevent to test custom code is preferable to use `createAsyncThunk` and `extraReducers` inside `citySlice` functions to handle effectively side effects in your dispatched actions